addOns/exim/src/main/java/org/zaproxy/addon/exim/har/HarImporter.java

```diff
@@ -118,29 +139,121 @@ private static void setHttpResponse(HarResponse harResponse, HttpMessage message
+        log.getEntries().removeAll(localPrivate);
+        int adjustment = h3ForRemoval.size() + localPrivate.size();
+        updateProgress(adjustment, "");
+        return log;
+    }
+
+    private void preProcessHarEntry(
+            List<HarEntry> h3ForRemoval, List<HarEntry> localPrivate, HarEntry entry) {
+        String vers = entry.getRequest().getHttpVersion();
+        preProcessH3Entries(h3ForRemoval, entry, vers);
+
+        String url = entry.getRequest().getUrl();
+        preProcessLocalPrivateEntries(localPrivate, entry, url);
+
+        // No point processing version or headers on entries being dropped
+        if (!localPrivate.contains(entry) && !h3ForRemoval.contains(entry)) {
+            preProcessEntryHttpVersion(entry, vers, false);
+            preProcessEntryHttpVersion(entry, entry.getResponse().getHttpVersion(), true);
+            entry.getResponse().getHeaders().forEach(this::preProcessHarHeaders);
+        }
+    }
+
+    private void preProcessH3Entries(List<HarEntry> h3ForRemoval, HarEntry entry, String vers) {
+        if (vers != null
+                && (entry.getRequest().getHttpVersion().equalsIgnoreCase("h3")
+                        || entry.getResponse().getHttpVersion().equalsIgnoreCase("h3"))) {
+            h3ForRemoval.add(entry);
+            LOGGER.warn(
+                    "Unsupported HTTP/3 (h3) message will be dropped: {}",
+                    entry.getRequest().getUrl());
+        }
+    }
+
+    private void preProcessLocalPrivateEntries(
+            List<HarEntry> localPrivate, HarEntry entry, String url) {
+        if (StringUtils.startsWithIgnoreCase(url, "about")
+                || StringUtils.startsWithIgnoreCase(url, "chrome")
+                || StringUtils.startsWithIgnoreCase(url, "edge")) {
+            localPrivate.add(entry);
+            LOGGER.debug("Skipping local private entry: {}", url);
+        }
+    }
+
+    private void preProcessEntryHttpVersion(HarEntry entry, String vers, boolean isResponse) {
+        if (vers == null || vers.isEmpty() || vers.equalsIgnoreCase(HttpHeader.HTTP)) {
+            if (isResponse) {
+                entry.getResponse().setHttpVersion(HttpHeader.HTTP11);
+            } else {
+                entry.getRequest().setHttpVersion(HttpHeader.HTTP11);
+            }
+            LOGGER.debug(
+                    "Setting {} version to {} for {}",
+                    isResponse ? "repsonse" : "request",
+                    entry.getRequest().getHttpVersion(),
+                    entry.getRequest().getUrl());
+        }
+    }
+
+    private void preProcessHarHeaders(HarHeader i) {
+        String name = i.getName();
+        String value = i.getValue();
+        if ((name.equalsIgnoreCase(HttpFieldsNames.CACHE_CONTROL)
+                        || name.equalsIgnoreCase(HttpFieldsNames.SET_COOKIE))
+                && (value.contains("\n") || value.contains("\r"))) {
+            i.setValue(i.getValue().replaceAll("[\r\n]", ", "));
```

Comment:
Should I just do this in all headers? I have no idea why the samples are like this, but these headers were where I encountered the issue 🤷‍♂️

Reply:
I'm not sure what to do here. It seems ZAP doesn't even handle that Set-Cookies might have multiple comma delimited cookies in a single header.
https://stackoverflow.com/a/2880070/7718222

Or if it does I haven't been able to figure out how.

Note: This issue was only encountered in really old har samples from Firefox 3.x the problem is actually `\u000a` which seems to be converted to `/n` by the library, which I'm then trying to change to comma...

Reply:
Not sure we should try salvage this case. Is this still seen in the wild? As for the parsing, we are following latest RFCs.

Reply:
For now I'll switch this to just drop them I guess. If people bring it up as an issue we can revisit.